docs/tutorial/03-resources.mdx

```diff
@@ -0,0 +1,488 @@
+  get your code running, but reading the rest is necessary to understand the
+  language's design.
+</Callout>
+
+This tutorial builds on the previous `Hello World` tutorial.
+Before beginning this tutorial, you should understand :
+
+- [Accounts](/cadence/language/accounts/)
+- [Transactions](/cadence/language/transactions/)
+- Signers
+- [Field types](/cadence/language/composite-types/)
+
+This tutorial will build on your understanding of accounts and how to interact with them by introducing resources.
+Resources are one of the Cadence's defining features.
+In Cadence, resources are a composite type like a struct or a class, but with some special rules.
+
+Resources are useful when you want to model direct ownership.
+Traditional structs or classes from other conventional programming languages are not an ideal way to represent ownership because they can be _copied_.
+This means a coding error can easily result in creating multiple copies of the same asset, which breaks the scarcity requirements needed for these assets to have real value.
+We have to consider loss and theft at the scale of a house, a car, or a bank account with millions of dollars, or a horse.
+Resources, in turn, solve this problem by making creation, destruction, and movement of assets explicit.
+
+In this tutorial, you will:
+
+1. Deploy a contract that declares a resource
+2. Save the resource into the account storage
+3. Interact with the resource we created using a transaction
+
+## Implementing a Contract with Resources
+
+---
+
+To interact with resources, you'll learn a few important concepts:
+
+- Using the create keyword
+- The move operator `<-`
+- The [Account Storage API](https://docs.onflow.org/cadence/language/accounts/#account-storage-api)
+
+Let's start by looking at how to create a resource with the `create` keyword and the move operator `<-`.
+
+You use the `create` keyword used to initialize a resource.
+Resources **must** be created before you can use them.
+
+The move operator `<-` is used to move a resource into a variable.
+You cannot use the assignment operator `=` with resources, so when you initialize a resource you will need to use the move operator `<-`.
+
+<Callout type="info">
+
+Open the Account `0x01` tab with file named `HelloWorldResource.cdc`. <br />
+`HelloWorldResource.cdc` should contain the following code:
+
+</Callout>
+
+```cadence:title=HelloWorldResource.cdc
+pub contract HelloWorld {
+
+    // Declare a resource that only includes one function.
+    pub resource HelloAsset {
+
+        // A transaction can call this function to get the "Hello, World!"
+        // message from the resource.
+        pub fun hello(): String {
+            return "Hello, World!"
+        }
+
+        // We're going to use the built-in create function to create a new instance
+        // of the HelloAsset resource
+        pub fun createHelloAsset(): @HelloAsset {
+            return <-create HelloAsset()
+        }
```

Comment:
I think the code here is at an incorrect location, `createHelloAsset()` should fall outside of the resource definition.